clients/fides-js/src/lib/hooks.ts

```diff
@@ -64,41 +65,79 @@ export const useDisclosure = ({ id }: { id: string }) => {
   };
 };
 
+/**
+ * Extracts the id value of each object in the list and returns a list
+ * of IDs, either strings or numbers based on the IDs' type.
+ */
+const extractIds = <T extends { id: string | number }[]>(
```

Comment:
can we reuse this function here? https://github.com/ethyca/fides/blob/main/clients/fides-js/src/components/tcf/TcfConsentButtons.tsx/#L15-L20 maybe extract it somewhere else, then share it between the two files?

(but also, nice use of `T`!)

Reply:
I thought about it but `getAllIds` casts the result to a string and for my use case, I need to preserve the original type

```
const getAllIds = (modelList: TcfModels) => {
  if (!modelList) {
    return [];
  }
  return modelList.map((m) => `${m.id}`);
};
```